src/DataProtection/DataProtection/src/KeyManagement/Key.cs

```diff
@@ -3,30 +3,156 @@
 
 using System;
 using System.Collections.Generic;
+using System.Xml.Linq;
 using Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption;
 using Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel;
+using Microsoft.AspNetCore.DataProtection.KeyManagement.Internal;
+using Microsoft.AspNetCore.DataProtection.XmlEncryption;
 
 namespace Microsoft.AspNetCore.DataProtection.KeyManagement;
 
 /// <summary>
-/// The basic implementation of <see cref="IKey"/>, where the <see cref="IAuthenticatedEncryptorDescriptor"/>
-/// has already been created.
+/// The basic implementation of <see cref="IKey"/>.
 /// </summary>
-internal sealed class Key : KeyBase
+internal sealed class Key : IKey
 {
+    private readonly Lazy<IAuthenticatedEncryptorDescriptor> _lazyDescriptor;
+    private readonly IEnumerable<IAuthenticatedEncryptorFactory> _encryptorFactories;
+
+    private IAuthenticatedEncryptor? _encryptor;
+
+    /// <summary>
+    /// The basic implementation of <see cref="IKey"/>, where the <see cref="IAuthenticatedEncryptorDescriptor"/>
+    /// has already been created.
+    /// </summary>
     public Key(
```

Comment:
I could be persuaded that the constructors should be private and there should be `Eager` and `Deferred` factory methods.